dnf/automatic/emitter.py

```diff
@@ -72,14 +78,20 @@ def notify_downloaded(self):
         assert self._available_msg
         self._downloaded = True
 
+    def notify_error(self, msg):
+        self._error = True
+        self._error_msg = msg
```

Comment:
Is it possible to have error == True, but no message?

Reply:
Not necessarily. It's possible to leave an empty string as the msg variable; however, when the _prepare_msg function is called, "An error has occurred on <system_name>" will be the first thing on the error message before appending the body. If needed, I can add a default value to self._error_msg but it would be a generic error message.